csharp/src/Apache.Arrow/Arrays/ArrayDataConcatenator.cs

```diff
@@ -151,6 +145,8 @@ public void Visit(UnionType type)
                 Result = new ArrayData(type, _totalLength, _totalNullCount, 0, buffers, children);
             }
 
+            public void Visit(MapType type) => ConcatenateLists(type.UnsortedKey());
```

Comment:
Technically I suppose there is a possibility the output map is still sorted (e.g. all the keys in the left map are less than the keys in the right map) but I agree this is the right call.  I don't know that there is any good spot to document this very minor note though.